test/vtgatev2_test.py

```diff
@@ -1210,20 +1210,10 @@ def test_restart_mysql_tablet_queries(self):
       t.wait_for_vttablet_state('SERVING')
     self.replica_tablet2.kill_vttablet()
     replica_tablet_proc = self.replica_tablet.kill_vttablet(wait=False)
-    time.sleep(0.1)
+    if vtgate_gateway_flavor().flavor() == 'shardgateway':
+      time.sleep(1)  # skip the vttablet waiting period
     # send query while vttablet is in lameduck, should fail as no vttablet
-    try:
-      vtgate_conn._execute(
-          'select 1 from vt_insert_test', {},
-          KEYSPACE_NAME, 'replica',
-          keyranges=[self.keyrange])
-      self.fail('DatabaseError should have been raised')
-    except Exception, e:  # pylint: disable=broad-except
-      self.assertIsInstance(e, dbexceptions.DatabaseError)
-      self.assertNotIsInstance(e, dbexceptions.IntegrityError)
-      self.assertNotIsInstance(e, dbexceptions.OperationalError)
-      self.assertNotIsInstance(e, dbexceptions.TimeoutError)
-    # send another query, should also fail
+    time.sleep(0.1)  # wait a short while so vtgate gets the health check
```

Comment:
We have a couple of other places where we poll for vars, with a timeout. I
think that would make sense here, just use a small timeout (maybe 0.2s for
this case).
On Feb 5, 2016 4:41 PM, "Liang" <notifications@github.com> wrote:

> In test/vtgatev2_test.py
> <https://github.com/youtube/vitess/pull/1486#discussion_r52092160>:
>
> >      # send query while vttablet is in lameduck, should fail as no vttablet
> > -    try:
> > -      vtgate_conn._execute(
> > -          'select 1 from vt_insert_test', {},
> > -          KEYSPACE_NAME, 'replica',
> > -          keyranges=[self.keyrange])
> > -      self.fail('DatabaseError should have been raised')
> > -    except Exception, e:  # pylint: disable=broad-except
> > -      self.assertIsInstance(e, dbexceptions.DatabaseError)
> > -      self.assertNotIsInstance(e, dbexceptions.IntegrityError)
> > -      self.assertNotIsInstance(e, dbexceptions.OperationalError)
> > -      self.assertNotIsInstance(e, dbexceptions.TimeoutError)
> > -    # send another query, should also fail
> > +    time.sleep(0.1)  # wait a short while so vtgate gets the health check
>
> There is. Though the case here is a bit different. We want to wait a bit
> to propagate the signal, but do not want to wait for too long. If we wait
> for the var to reduce, it would not detect the bug that the signal failed
> to propagate in a reasonable amount of time (milliseconds).
> I do not like sleep, so I am happy to take suggestions to improve it.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/youtube/vitess/pull/1486/files#r52092160>.
>